### PR TITLE
Feature: add swagger document

### PR DESCRIPTION
**Pull request by zh9310**
add swagger document

### CODE REVIEW

File: src/api-server/cmd/docs/swagger.json [hunk 1]
```diff
@@ -0,0 +1,349 @@
+{
```
Comment (nascentcore-eng) [inline src/api-server/cmd/docs/swagger.json:1]: Is this file automatically generated?

File: src/api-server/cmd/main.go [hunk 2]
```diff
@@ -62,9 +63,19 @@ var (
 	)
 )
 
+// @contact.name   API Support
+// @license.name  Apache 2.0
```
Comment (nascentcore-eng) [inline src/api-server/cmd/main.go:67]: We use AGPL, not Apache, please check the repo root for the actual license version

File: src/api-server/cmd/main.go [hunk 3]
```diff
@@ -62,9 +63,19 @@ var (
...
+// @license.url   http://www.apache.org/licenses/LICENSE-2.0.html
 func main() {
 	flag.Parse()
 
+	docs.SwaggerInfo.Title = "Swagger Example API"
+	docs.SwaggerInfo.Description = "This is a sample server Petstore server."
+	docs.SwaggerInfo.Version = "1.0"
+	docs.SwaggerInfo.Host = "petstore.swagger.io"
+	docs.SwaggerInfo.BasePath = "/v2"
+	docs.SwaggerInfo.Schemes = []string{"http", "https"}
```
Comment (nascentcore-eng) [inline src/api-server/cmd/main.go:77]: These are completely wrong...

File: src/api-server/http/http.go [hunk 4]
```diff
@@ -60,6 +63,7 @@ func StartHTTPService(cfg Config, pgClient *pg.Client) {
 
 	addr := fmt.Sprintf(":%d", cfg.Port)
 	log.Infof("Listening on %s ...", addr)
+	router.GET("/swagger/*any", ginSwagger.WrapHandler(swaggerFiles.Handler))
```
Comment (nascentcore-eng) [inline src/api-server/http/http.go:66]: Document in the comment about the purpose of this

File: src/api-server/http/module_manager.go [hunk 5]
```diff
@@ -28,6 +28,15 @@ type ModuleManager struct {
 	PGClient      *pg.Client
 }
 
+// ShowAccount godoc
+// @Summary      Add module
+// @Description  Create Module
+// @Tags         module
+// @Accept       json
+// @Produce      json
+// @Param			   account	body		module.Module	true	"Add module"
```
Comment (nascentcore-eng) [inline src/api-server/http/module_manager.go:37]: Why there is long spaces separating `@param` & `account`

File: src/api-server/http/module_manager.go [hunk 6]
```diff
@@ -87,6 +96,15 @@ func (mgr *ModuleManager) createModule(c *gin.Context) {
 	c.JSON(http.StatusOK, gin.H{"code": "200", "message": "create success, module id: " + mod.ID})
 }
 
+// ShowAccount godoc
+// @Summary      List all module
+// @Description  Create Module
+// @Tags         module
+// @Accept       json
+// @Produce      json
+// @Param			fields	query		string	false	"query field search like 'id,name,createTime'"
```
Comment (nascentcore-eng) [inline src/api-server/http/module_manager.go:105]: Same here

File: src/api-server/cmd/docs/README.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+## create swagger
+- use swag cli create docs[swag](https://pkg.go.dev/github.com/go-openapi/swag)
+```shell
+cd starship/src/api-server/cmd
+swag init -g ./api-server/http/http.go -d ../../
+```
+
+## Visit document
+http://host:port/swagger/index.html
```
Comment (nascentcore-eng) [inline src/api-server/cmd/docs/README.md:11]: Describe how this module works, i.e., the mechanism of swagger generating docs.

File: src/api-server/cmd/docs/docs.go [hunk 8]
```diff
@@ -0,0 +1,376 @@
+// Code generated by swaggo/swag. DO NOT EDIT
```
Comment (nascentcore-eng) [inline src/api-server/cmd/docs/docs.go:1]: Why place docs under src/api-server/cmd?

src/api-server/cmd is for storing executables, docs are not exectuable, right?

File: src/api-server/cmd/docs/swagger.yaml [hunk 9]
```diff
@@ -0,0 +1,234 @@
+definitions:
```
Comment (nascentcore-eng) [inline src/api-server/cmd/docs/swagger.yaml:1]: Is this automatically generated?

File: src/api-server/cmd/docs/README.md [hunk 10]
```diff
@@ -0,0 +1,11 @@
+# API Server http document
+
+## create swagger
+- use swag cli create docs[swag](https://pkg.go.dev/github.com/go-openapi/swag)
+```shell
+cd starship/src/api-server/cmd
+swag init -g ./api-server/http/http.go -d ../../
+```
+
```
Comment (nascentcore-eng) [inline src/api-server/cmd/docs/README.md:9]: Also need to explain the purpose of .json and .yaml files under this directory.

**Review by nascentcore-eng — CHANGES_REQUESTED** [review]

**Comment by zh9310** [comment]
Too many backward main branches

File: src/api-server/cmd/docs/swagger.json (hunk 1 above)
Comment (zh9310) [reply src/api-server/cmd/docs/swagger.json:1]: This file is not required.
https://github.com/tricorder-observability/starship/pull/65

**Pull request closed without merging**